cloudinit/sources/azure/errors.py

```diff
@@ -150,13 +151,37 @@ def __init__(self, *, exception: UrlError, duration: float) -> None:
         self.supporting_data["url"] = exception.url
 
 
+class ReportableErrorImdsInvalidMetadata(ReportableError):
```

Comment:
this error class was already added in this PR: https://github.com/canonical/cloud-init/pull/4596/files#diff-7bb4ccc9a8918f60539680c6ef402206007fc774320acdad3dc07c2ca791fa42R175